### PR TITLE
Detect supported modules when running the tests

### PR DESCRIPTION
**Pull request by galpeter**
The patch will allow the test framework to automatically
detect which modules are built-in the IoT.js. Based on this
information it can check if the input test file contains
one of the supported module name and if not the test can
be skipped.

IoT.js-DCO-1.0-Signed-off-by: Peter Gal pgal.u-szeged@partner.samsung.com

### CODE REVIEW

File: tools/test_runner.js [hunk 1]
```diff
@@ -15,6 +15,8 @@
 
 var testdriver = require('testdriver');
 var console_wrapper = require('common_js/module/console');
+var builtin_modules = ['process'].concat(Object.keys(process.native_sources),
```
Comment (hs0225) [inline tools/test_runner.js]: `process` is already included `process.native_sources`.

Comment (galpeter) [inline tools/test_runner.js]: True, will fix asap.

**Comment by galpeter** [comment]
@hs0225 I've update the pr.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**